### PR TITLE
adding more actions to BTCPP executor

### PR DESCRIPTION
**Pull request by facontidavide**
# Additions

Works as expected: 

- PickObject
- PlaceObject
- Detect Object

Don't work (yet):

- OpenDoor
- CloseDoor

I get the message:

```
[OpenDoor] failed with error: PRECONDITION_FAILURE. Message: Robot is not at an openable location.
```
This is the sim state:
image

So, I guess that I need to be in front of the door, but that location has no name.
@sea-bass con you suggest the correct way to deal with this?

### CODE REVIEW

**Comment by dcconner** [comment]
Minor tweak in CMakeLists, and request to add export include:

```bash
diff --git a/technologies/BehaviorTree.CPP/pyrobosim_btcpp/CMakeLists.txt b/technologies/BehaviorTree.CPP/pyrobosim_btcpp/CMakeLists.txt
index bd47242..b924281 100644
--- a/technologies/BehaviorTree.CPP/pyrobosim_btcpp/CMakeLists.txt
+++ b/technologies/BehaviorTree.CPP/pyrobosim_btcpp/CMakeLists.txt
@@ -38,7 +38,12 @@ install(
 
 install(
   TARGETS btcpp_executor
-  DESTINATION DESTINATION lib/${PROJECT_NAME}
+  DESTINATION lib/${PROJECT_NAME}
 )
 
+install(DIRECTORY include/  DESTINATION include/)
+
+ament_export_dependencies(${THIS_PACKAGE_DEPS})
+ament_export_include_directories(include)
+
 ament_package()
```

**Comment by dcconner** [comment]
2 comments:

Should the trees have unique names so we can access via `ExecuteTree` action ?
Currently the demos all have  `<BehaviorTree ID="MainTree">` as the ID.

I posted an issue https://github.com/BehaviorTree/BehaviorTree.ROS2/issues/97 that impacts ability to use `ExecuteTree`

File: technologies/BehaviorTree.CPP/pyrobosim_btcpp/trees/problem1.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            <DetectObject name="DetectSnack" object="snacks0"/>
+            <Sequence name="DetectedTrue">
+                <PickObject name="PickSnack" object="snacks0"/>
+                <Navigate name="ToTableWithSnack" target="table"/>
+                <PlaceObject name="PlaceSnack" object="snacks0"/>
+            </Sequence>
+            <Navigate name="ToTableWithoutSnack" target="table"/>
+        </IfThenElse>
+        <Sequence name="DetecSnackBackToPantry">
+            <PickObject name="PickSnack" object="snacks0"/>
```
[the remark this answers is not in the document]
Comment (facontidavide) [reply technologies/BehaviorTree.CPP/pyrobosim_btcpp/trees/problem1.xml:16]: yes, I was just playing around

**Comment by facontidavide** [comment]
@dcconner thanks. changes applied

**Pull request merged**